File: app/src/main/java/com/hazizz/droid/Communication/POJO/Response/CommentSectionPOJOs/POJOGroup.java
package com.hazizz.droid.Communication.POJO.Response.CommentSectionPOJOs;


import com.hazizz.droid.Communication.POJO.Response.Pojo;

import lombok.Data;

@Data
public class POJOGroup implements Pojo {
    private int id;
    private String name;
    private String groupType;
    private int userCount;

    public POJOGroup(int id, String name, String groupType, int userCount){
        this.id = id;
        this.name = name;
        this.groupType = groupType;
        this.userCount = userCount;
    }
}

File: lib/communication/pojos/PojoGroup.dart
import 'package:json_annotation/json_annotation.dart';

import 'Pojo.dart';
part 'PojoGroup.g.dart';


@JsonSerializable()
class PojoGroup extends Pojo {
  int id;
  String name;
  String uniqueName;
  String groupType;
  int userCount;

  PojoGroup(this.id, this.name, this.uniqueName, this.groupType,
      this.userCount);

  factory PojoGroup.fromJson(Map<String, dynamic> json) =>
      _$PojoGroupFromJson(json);

  Map<String, dynamic> toJson() => _$PojoGroupToJson(this);

}

@JsonSerializable()
class PojoGroupWithoutMe extends PojoGroup {
  int id;
  String name;
  String uniqueName;
  String groupType;
  int userCount;
  int userCountWithoutMe;

  PojoGroupWithoutMe(this.id, this.name, this.uniqueName, this.groupType, this.userCount, this.userCountWithoutMe) : super(id, name, uniqueName, groupType, userCount);

  factory PojoGroupWithoutMe.fromJson(Map<String, dynamic> json) =>
      _$PojoGroupWithoutMeFromJson(json);

  Map<String, dynamic> toJson() => _$PojoGroupWithoutMeToJson(this);

}